src/validate.ts

```diff
@@ -74,6 +82,49 @@ export function validate(
     return [];
   }
 
+  function validateFunctionNode(node: FunctionInterface) {
+    const func =
+      node.arguments.length === 4 ? node.arguments[3] : node.arguments[2];
+
+    return [
+      // visit all other arguments
+      ...node.arguments
+        .filter((arg) => arg !== func)
+        .flatMap((arg) => validateEachChildRecursive(arg, visit)),
+      // process the function closure
+      ...validateEachChildRecursive(func, validateFunctionClosureNode),
```

Comment:
This is the only line that is specific to this function

Reply:
yes, but visitEachChild doesn't support ignoring a specific child and we are looking for the `new Function` expression with arguments. So we are picking out the `new`, finding the special function argument, validating it separately and carrying on the "forEach" with the remaining arguments.

I updated step functions to do the same after finding the bugs in here: https://github.com/functionless/functionless/pull/243#discussion_r899402510

Reply:
The other options is that we check each function argument to see if it has a parent of type function/step function and that its the right position in the arguments.  I kind of like this because it would be easier to refactor how a step function behavior or add more cases.